core/consensus/babe/babe_lottery.hpp

```diff
@@ -32,14 +33,18 @@ namespace kagome::consensus {
 
     /**
      * Compute leadership for all slots in the given epoch
-     * @param epoch structure; "epoch_index" MUST be set to the epoch, for which
-     * the randomness is to be computed
-     * @param keypair of this node
+     * @param randomness is a seed to compute leadership
+     * @param threshold is a maximum value that is considered valid by vrf
+     * @param epoch_length is the length of resulting array (each element
+     * corresponds to randomness in a slot)
      * @return vector of outputs; none means the peer was not chosen as a leader
      * for that slot, value contains VRF value and proof
      */
     virtual SlotsLeadership slotsLeadership(
-        const Epoch &epoch, crypto::SR25519Keypair keypair) const = 0;
+        const Randomness &randomness,
+        const Threshold threshold,
+        EpochIndex epoch_length,
```

Comment:
Is incorrect type name or type. May be better use size_t?

Reply:
It is an alias on size_t. If we need to change the type of EpochIndex we would change just single line. So better to stick with this approach